python/ray/rllib/agents/sac/sac_policy_graph.py

```diff
@@ -169,35 +168,39 @@ def _get_Q_targets(self):
         next_log_pis = self.policy.log_pis(self._next_observations_ph,
                                            next_actions)
 
-        next_Q_values = tf.stack([
+        next_Qs_values = tf.stack([
             Q_target([self._next_observations_ph, next_actions])
             for Q_target in self.Q_targets
         ])
 
-        next_values = next_Q_values - self.alpha * next_log_pis[None, :, :]
-        assert next_values.shape.as_list() == [2, None, 1]
+        min_next_Q_values = tf.reduce_min(next_Qs_values, axis=0)
+        next_values = min_next_Q_values - self.alpha * next_log_pis
+        assert next_values.shape.as_list() == [None, 1]
 
-        discount = self.config["gamma"]**self.config["n_step"]
+        discount = self.config["gamma"] ** self.config["n_step"]
         # td target
-        return (self._rewards_ph[None, :, None] + discount *
-                (1.0 - tf.to_float(self._terminals_ph[None, :, None])) *
+
+        return (self._rewards_ph[..., None] + discount *
+                (1.0 - tf.to_float(self._terminals_ph[..., None])) *
                 next_values)
 
     def _init_critic_loss(self):
         Q_targets = tf.stop_gradient(self._get_Q_targets())
 
-        assert Q_targets.shape.as_list() == [2, None, 1]
+        assert Q_targets.shape.as_list() == [None, 1]
 
         Q_values = self.Q_values = tf.stack(
             [Q([self._observations_ph, self._actions_ph]) for Q in self.Qs])
 
         assert Q_values.shape.as_list() == [2, None, 1]
 
         Q_loss_weight = self.config["optimization"]["Q_loss_weight"]
-        self.td_error = tf.reduce_mean(Q_targets - Q_values, axis=0)
+        self.td_error = tf.reduce_mean(Q_targets[None, ...] - Q_values)
```

Comment:
I think you dropped the axis=0. It's supposed to be one td error per update example for it to be used in computing td errors for prioritized experience replay. Not that you would want to use it with soft actor-critic, but I still think it might as well be supported since it's so easy to support.

Reply:
Ah, good catch. I'll check it out tomorrow and make sure it's computed correctly.